reactor-core/src/main/java/reactor/core/publisher/FluxOnAssembly.java

```diff
@@ -378,17 +370,29 @@ final public int requestFusion(int requestedMode) {
 
 		final Throwable fail(Throwable t) {
 			StringBuilder sb = new StringBuilder();
-			fillStacktraceHeader(sb, parent.getClass(), snapshotStack);
 			OnAssemblyException set = null;
-			if (!snapshotStack.isLight()) {
+			final String backtrace;
```

Comment:
the `backtrace` could be made to be `snapshotStack.toAssemblyInformation()` in all cases if the `isLight` test is done in the publisher as well as `Traces.extractOperatorAssemblyInformation(sb.toString(), true)` being done early and only if not light, wdyt?

Reply:
But `toAssemblyInformation` only returns the info, while `backtrace` also contains the header